Ardupilog.m

```diff
@@ -427,41 +419,47 @@ function delete(obj)
             slice.numMsgs = 0;
             
             % Loop through the LogMsgGroups, slicing each one
-            for msgGroup = slice.logMsgGroups
-                % If the LogMsgGroup has been deleted, skip it
-                if ~isvalid(msgGroup)
-                    continue
+            logProps = properties(obj);
+            for i = 1:length(logProps)
+                propertyName = logProps{i}; % Get the name of the property under examination
+                % We are interested only in LogMsgGroup objects, skip the rest of the properties
+                if ~isa(obj.(propertyName),'LogMsgGroup') 
+                    continue;
                 end
                 
                 % Slice the LogMsgGroup
-                lmg_slice = slice.(msgGroup.Name).getSlice(slice_values, slice_type);
+                lmg_slice = slice.(propertyName).getSlice(slice_values, slice_type);
                 % If the slice is not empty, add it to the Ardupilog slice
                 if isempty(lmg_slice)
-                    delete(msgGroup)
+                    delete(slice.(propertyName))
                 else
-                    slice.(msgGroup.Name) = lmg_slice;
-                    slice.numMsgs = slice.numMsgs + size(slice.(msgGroup.Name).LineNo, 1);
+                    slice.(propertyName) = lmg_slice;
+                    slice.numMsgs = slice.numMsgs + size(slice.(propertyName).LineNo, 1);
                 end
             end
         end
         
     end
+    
     methods(Access=protected)
-        function cpObj = copyElement(obj)
-        % Makes copy() into a "deep copy" method (i.e. when copying an Ardupilog, the
-        % new copy also has all the data stored in dynamic-property LogMsgGroups)
-            
-            % Create a standard copy (to copy all properties)
-            cpObj = copyElement@matlab.mixin.Copyable(obj);
-            
-            % Deep-copy the Dynamic Properties
-            for ndx = 1:length(obj.logMsgGroups)
-                % Create a new dynamic property
-                cpObj.logMsgGroups(ndx) = addprop(cpObj, obj.logMsgGroups(ndx).Name);
-                % Copy the data from the original
-                cpObj.(obj.logMsgGroups(ndx).Name) = copy(obj.(obj.logMsgGroups(ndx).Name));
+        
```

Comment:
This new copy functionality seems to work great! I think it's also a bit clearer... I think we should make a nearly identical change to LogMsgGroup.copyElement.